seed/seed20.go

```diff
@@ -580,9 +580,15 @@ func (s *seed20) doLoadMeta(handler SnapHandler, tm timings.Measurer) error {
 	}
 	stopCh := make(chan struct{})
 	outcomesCh := make(chan error, njobs)
+	var wg sync.WaitGroup
+	wg.Add(njobs)
 	for j := 1; j <= njobs; j++ {
 		jtm := tm.StartSpan(fmt.Sprintf("do-load-meta[%d]", j), fmt.Sprintf("snap metadata loading job #%d", j))
 		go func() {
+			// defers are LIFO, make sure that time snap is stopped
+			// before we let the parent know that the goroutine is
+			// done
+			defer wg.Done()
 			defer jtm.Stop()
```

Comment:
shouldn't we instead move the Stop before we send things on the channels? it's a bit of duplication, but adding a wait group just for dealing with timings seems also a bit overkill?

Reply:
That would seem to solve the problem as well. The wait group is more explicit as to why it is safe, but moving the `jtm.Stop()` call is lighter-weight.